### DISCUSSION TITLE
How to enable both Vim Plug and Lazy.vim for plugin management?

### DISCUSSION THREAD

**Discussion by uwla** [LazyVim/LazyVim#4282] [Q&A]
Hello. I want to migrate from Vim Plug to Lazy.vim, but there are so many plugins and options that would require significant effort to migrate all at once. Is there a way I can use Vim Plug and LayzVim simultaneously, so I can migrate the plugins and their configs to LazyVim gradually?

**Comment by dpetka2001** [comment]
I don't believe you can use them both together at the same time. What you can do is take advantage of `:h $NVIM_APPNAME` Neovim's feature, so that you can have 2 different configs at the same time, one with your current configuration and one based on `lazy.nvim` until you decide which one will be your main.

**Reply by uwla** [reply]
I see...
Could you suggest some link to migrate from VimPlug to LazyVim?
I do not know how I would migrate to lazyvim my `vimscript` plugins whose configuration is defined via `let g:plugin_var` and also some `lua` plugins (such as lsp, telescope, nvim-cmp) that have lots of function declarations in them...

**Reply by dpetka2001** [reply]
Did you read the help page for `:h $NVIM_APPNAME`? It allows you to have multiple configurations in parallel so you can test out things.
Basically you can do 
```bash
1. mkdir ~/.config/lazytest
2. git clone https://github.com/LazyVim/starter ~/.config/lazytest
3. NVIM_APPNAME=lazytest nvim and wait for the installation to complete
4. While Lazy UI is still open press `C` to check and `U` to make sure everything is updated
5. Restart Neovim with command from step 3
```
Every time now you want to try LazyVim you will be invoking `nvim` with the command from step 3 (you can just make an `alias` for it).
For your Vimscript plugins instead of `let g:plugin_var` you can do `vim.g.plugin_var` in the `init` function in the plugin spec. Rest of Lua plugins (telescope, nvim-cmp etc) you might have to make some minor adjustments to the way LazyVim's structure of plugins is configured. 

I don't know if there's an easy migration guide. I also used VimPlug in the past, but for a really brief time and then just started fresh from scratch when I tried LazyVim. If you want to do a full migration of your current setup it will definitely take some time. That's why I suggested the above steps, so you can have both your current setup and LazyVim setup in 2 different directories and be able to slowly migrate things over at your own pace. It would be good to read the LazyVim [website docs](https://www.lazyvim.org/) to get a touch of familiarity first about how things are done in LazyVim.

**Comment by Steve-V** [comment]
Haven't tried this myself but it may be what you're looking for:
[lazy.nvim adapter for vim-plug](https://gist.github.com/BlueDrink9/474b150c44d41b80934990c0acfb00be)
